src/sensitivity_interface.jl

```diff
@@ -67,17 +69,94 @@ instead of just at discrete data points.
+```
+
+As defined, that cost function only has non-zero values over nontrivial intervals. However, in many
+cases one may want to include in the cost function loss values at discrete points, for example, matching
+the data at time points `t`. In this case, terms of `g` can be represented by Dirac delta functions
+which are then applied in the corresponding ``\lambda^\star`` and ``\frac{dG}{dp}`` equations. 
+
+For more information, see [Sensitivity Math Details](@ref sensitivity_math).
+
+## Positional Arguments
+
+- `sol`: the solution from the forward pass of the ODE. Note that if not using a checkpointing
+  sensitivity algorithm, then it's assumed that the (dense) interpolation of the forward solution is
+  of sufficient accuracy for recreating the solution at any time point.
+- `alg`: the algorithm (i.e., DiffEq solver) to use for the solution of the adjoint problem.
+
+## Keyword Arguments
+
+- `t`: the time points at which the discrete cost function is to be evaluated.
+  This argument is only required if discrete cost functions are declared.
+- `g`: the continuous instantaneous cost ``g(u,p,t)`` at a given time point
+  represented by a Julia function `g(u,p,t)`. This argument is only required
+  if there is a continuous instantaneous cost contribution.
+- `dgdu_discrete`: the partial derivative ``g_u`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdu_discrete`
+  is required.
+- `dgdp_discrete`: the partial derivative ``g_p`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdp_discrete`
+  is not required and is assumed to be zero.
+- `dgdu_continuous`: the partial derivative ``g_u`` evaluated at times
+  not corresponding to terms with an associated Dirac delta. If `g` is given,
+  then this term is not required and will be approximated by numerical or (forward-mode) automatic
+  differentiation (via the `autodiff` keyword argument in the `sensealg`)
+  if this term is not given by the user.
+- `dgdp_continuous`: the partial derivative ``g_p`` evaluated at times
+  not corresponding to terms with an associated Dirac delta. If `g` is given,
+  then this term is not required and will be approximated by numerical or (forward-mode) automatic
+  differentiation (via the `autojacvec` keyword argument in the `sensealg`)
+  if this term is not given by the user.
+- `abstol`: the absolute tolerance of the adjoint solve. Defaults to `1e-3`
+- `reltol`: the relative tolerance of the adjoint solve. Defaults to `1e-3`
+- `checkpoints`: the values to use for the checkpoints of the reverse solve, if the
+  adjoint `sensealg` has `checkpointing = true`. Defaults to `sol.t`, i.e. the
+  saved points in the `sol`.
+- `corfunc_analytical`: the function for the conversion from an Ito to a Stratanovich definition of an SDE, i.e.
+  ``- 1/2 b_u b`` for ``dX = a(X,t)dt + b(X,t)dW_t``. Only used if the `sol.prob isa SDEProblem`. If not given, this is
```

Comment:
```suggestion
For sensitivity analysis of an SDE in the Ito sense ``dX = a(X,t)dt + b(X,t)dW_t`` with conversion term ``- 1/2 b_X b``, `corfunc_analytical` denotes `b_X b``.
Only used if the `sol.prob isa SDEProblem`. If not given, this is
```

Reply:
it makes sense, just needed docs 😅

Reply:
So okay, why are we using reverse mode here? Looks like a Jv operation?

Reply:
oh wait -- sorry about that (I confused the notation)

it's 
```
(b(X_t) · ∇_X) b(X_t)
```